### PR TITLE
[12.0] Migrate account_netting to v12

### PR DESCRIPTION
**Pull request by alexis-via**

### CODE REVIEW

**Comment by alexis-via** [comment]
I also converted the readme to the new subdir format.

File: account_netting/readme/CONTRIBUTORS.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+* Pedro M. Baeza <pedro.baeza@tecnativa.com>
+* Vicent Cubells <vicent.cubells@serviciosbaeza.com>
```
Comment (pedrobaeza) [inline account_netting/readme/CONTRIBUTORS.rst:2]: This email address is incorrect, but any way, we lastly prefer to remove our addresses from here. And include yourself here.

**Review by pedrobaeza — APPROVED** [review]

**Review by tbaden — APPROVED** [review]

File: account_netting/readme/CONTRIBUTORS.rst (hunk 1 above)
Comment (pedrobaeza) [reply account_netting/readme/CONTRIBUTORS.rst:2]: @alexis-via can you perform this change and I merge?

**Review by cfreyse — APPROVED** [review]

**Comment by kittiu** [comment]
From the functional test, I found possible bug as follow.

- I created 2 invoice, which result in AP 200.00 and AR 300.00
- I then do the netting this 2 move line, The AP 200 will be fully reconciled, while AR 300 is partially reconciled (with balance = 100). On the wizard it shows this -> OK

image

- I created another payable invoice, which result in AP 10.0
- I then do the netting again for this AR 300 (remain 100) and AP 10, I am expecting the wizard to show Balance only 90.00 But it is showing 290.00 - NOT OK

image

In the code, I see that the balance is calculated from debit/credit, and not from amount_residual

In Summary, I think currently it is correct only if all the move line are fully unreconciled yet. If any line is partially reconciled, it can be wrong.

**Comment by rousseldenis** [comment]
> n Summary, I think currently it is correct only if all the move line are fully unreconciled yet. If any line is partially reconciled, it can be wrong.

@kittiu Do you have time to make a PR on akretion repo to fix this ?

**Comment by kittiu** [comment]
Ok. I will check it out.

**Comment by kittiu** [comment]
@rousseldenis I have the code fixed locally. What do you mean when saying "PR on akretion repo"?
I do not have authority to branch from akretion and make PR.
And if I want to fork again, I also can't since I already have one fork.
Or should I do PR base off my fork? (which is not a good idea). Any suggestion?

**Comment by rousseldenis** [comment]
@kittiu I've discussed with @alexis-via yesterday about that and the use case you mention (on partial reconciliation) is not a common one.

Either we conclude it's a limitation of this module, either we include the case in the code.

To do a PR on akretion repo is quite easy (to make your modifications included in this):

- Make a branch, make your modifs, commit
- Then Push to your repo and do a PR changing the origin branch to akretion:12-mig-netting
Then wait for answer to make it merged in this

**Comment by kittiu** [comment]
@rousseldenis sorry for dumb question, how to rebase on this branch? Now, I only have the fork from OCA/account-financial-tools (which I can not delete otherwise, my other work is gone).

Or can I submit as patch file?
[patch.txt](https://github.com/OCA/account-financial-tools/files/3231368/patch.txt)

**Comment by rousseldenis** [comment]
> @rousseldenis sorry for dumb question, how to rebase on this branch? Now, I only have the fork from OCA/account-financial-tools (which I can not delete otherwise, my other work is gone).
> 
> Or can I submit as patch file?
> [patch.txt](https://github.com/OCA/account-financial-tools/files/3231368/patch.txt)

Ok, it's a little bit tricky to do that but that allows to easily add changes to an existing PR if the original author as no time to change it or if you have the solution already.

I will give you the steps to do it:

- Add a remote for akretion : git remote add akretion https://github.com/akretion/account-financial-tools
- git fetch akretion
- Make a branch based on the PR one : git checkout -b your-branch akretion/akretion:12-mig-netting
- Commit your changes
- Push to your repo
- Make the PR based on  akretion/akretion:12-mig-netting

I hope it's clear :smile:

**Comment by kittiu** [comment]
@rousseldenis it works now :+1: https://github.com/akretion/account-financial-tools/pull/7

Just note it here for future reference :)
Here is how to make new branch out of 3rd party branch (akretion/12-mig-netting) in to my own fork

- Add a remote for akretion : git remote add akretion https://github.com/akretion/account-financial-tools
- git fetch akretion 12-mig-netting
- Make a branch based on the PR one : git checkout -b <new-branch> akretion/12-mig-netting
- Commit your changes
- Push to your repo > git push origin <new-branch>
- Make the PR based on akretion:12-mig-netting

This is why I love OCA :-)

File: account_netting/readme/USAGE.rst [hunk 2]
```diff
@@ -0,0 +1,12 @@
+From any account journal entries view:
+
+* Accounting/Journal Entries/Journal Items
```
Comment (Tardo) [inline account_netting/readme/USAGE.rst:3]: Please, update this with the correct way

**Review by Tardo — CHANGES_REQUESTED** [review]

**Review by Tardo — APPROVED** [review]
LGTM, tested on runbot. The documentation can be improved.
